src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_deep_graph.php

```diff
@@ -85,16 +85,12 @@ protected function getFooService()
         if (isset($this->services['foo'])) {
             return $this->services['foo'];
         }
-
         $b = new \c2();
-
-        $this->services['foo'] = $instance = new \c1($a, $b);
-
         $c = new \c3();
-
         $c->p3 = new \c4();
+
         $b->p2 = $c;
 
-        return $instance;
+        return $this->services['foo'] = new \c1($a, $b);
```

Comment:
The effect of the fix is visible here: $b is now fully configured when it is injected, where previously its p2 property was set after the injection.